webapp/channels/src/components/widgets/menu/menu_modals/submenu_modal/submenu_modal.tsx

```diff
@@ -17,63 +16,61 @@ type Props = {
     onExited: () => void;
 }
 
-type State = {
-    show: boolean;
-}
-
-export default class SubMenuModal extends React.PureComponent<Props, State> {
-    public constructor(props: Props) {
-        super(props);
-        this.state = {
-            show: true,
-        };
-    }
+const SubMenuModal = ({
+    elements,
+    onExited,
+}: Props) => {
+    const [show, setShow] = useState(true);
+    const intl = useIntl();
 
-    public onHide = () => { //public because it is used on tests
-        this.setState({show: false});
-    };
+    const onHide = useCallback(() => {
+        setShow(false);
+    }, [setShow]);
```

Comment:
`setShow` is a useState setter, so it won't change between renders. It is safe to remove it from here.

Reply:
Done